packages/iris-grid/src/IrisGridTableModelTemplate.ts

```diff
@@ -1552,7 +1552,11 @@ class IrisGridTableModelTemplate<
       range.startRow <
         this.floatingTopRowCount + this.table.size + this.pendingRowCount &&
       range.endRow <
-        this.floatingTopRowCount + this.table.size + this.pendingRowCount
+        this.floatingTopRowCount + this.table.size + this.pendingRowCount &&
+      (this.inputTable.keyColumns.length !== 0 ||
+        (this.inputTable.keyColumns.length === 0 &&
+          this.isPendingRow(range.startRow) &&
+          this.isPendingRow(range.endRow)))
```

Comment:
Nice, this fixes it, but I think we could re-write this a little bit so it's a little more readable.
On line `1548` there's already the `this.isPendingRow(range.startRow) && this.isPendingRow(range.endRow)` check. I would write this logic into that part into there, e.g.:
```
      ((this.isPendingRow(range.startRow) && this.isPendingRow(range.endRow)) ||
        (this.inputTable.keyColumns.length > 0 && range.startColumn >= this.inputTable.keyColumns.length &&
range.endColumn >= this.inputTable.keyColumns.length)) &&
```

Side note, because of how large this conditional has become, it would almost make more sense to just expand it out so it's not so hard on the brain, something like:
```
isEditableRange(range: GridRange): boolean {
  // Make sure we have an input table and a valid range
  if (this.inputTable == null || !GridRange.isBounded(range)) {
    return false
  }
  
  // Check that the edit is in the editable range
  // If an input table has keyed columns, the non-key columns are always editable
  // If an input table does not have key columns, it is append only and existing rows cannot be editable
  // Pending rows are always editable
  if (...) {
    return false;
  }
  
  // Editing the aggregations/totals which are floating above/below is not allowed
  if (...) {
    return false;
  }
  
  return true;
}
```